spec-tests/src/test/java/org/candlepin/spec/PoolSpecTest.java

```diff
@@ -0,0 +1,673 @@
+     *  The base string to concatenate to the generated suffix
+     *
+     * @return
+     *  the provided base string/prefix concatenated with a randomly generated suffix
+     */
+    private String randomSuffix(String prefix) {
+        return prefix + "-" + ((int) (Math.random() * 100000));
+    }
+
+    private ApiClient createUserClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        UserDTO user = UserUtil.createUser(client, owner);
+        return ApiClients.trustedUser(user.getUsername(), true);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        String consumerName = this.randomSuffix("test_consumer");
+        ConsumerDTO consumer = this.createConsumer(client, owner, consumerName, "system", null);
+
+        return this.createConsumerClient(client, consumer);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, ConsumerDTO consumer) throws ApiException {
+        return ApiClients.ssl(consumer.getIdCert());
+    }
+
+    private OwnerDTO createOwner(ApiClient client) throws ApiException {
+        return client.owners().createOwner(Owners.random());
+    }
+
+    private ConsumerDTO createConsumer(ApiClient client, OwnerDTO owner, String name, String type,
+        Map<String, String> facts) throws ApiException {
+
+        ConsumerTypeDTO ctype = new ConsumerTypeDTO()
+            .label(type);
+
+        ConsumerDTO consumer = new ConsumerDTO()
+            .name(name)
+            .owner(Owners.toNested(owner))
+            .type(ctype);
+
+        if (facts != null) {
+            // Copy the map into a new hashmap to ensure we can still modify it later if we so choose
+            consumer.facts(new HashMap<>(facts));
+        }
+
+        consumer.putFactsItem("system.certificate_version", "3.3");
+
+        return client.consumers()
+            .register(consumer);
+    }
+
+    private ProductDTO createProduct(ApiClient client, OwnerDTO owner) throws ApiException {
+        String productId = String.valueOf((int) Math.random() * 10000);
+
+        ProductDTO product = new ProductDTO()
+            .id(productId)
+            .name("test_product-" + productId);
+
+        return client.ownerProducts()
+            .createProductByOwner(owner.getKey(), product);
+    }
+
+    // ugh...
+    private AttributeDTO buildAttribute(String key, String value) {
+        return new AttributeDTO()
+            .name(key)
+            .value(value);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner) throws ApiException {
+        ProductDTO product = this.createProduct(client, owner);
+        return this.createPool(client, owner, product);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner, ProductDTO product) throws ApiException {
+        OffsetDateTime startDate = OffsetDateTime.now();
+        OffsetDateTime endDate = startDate.plusYears(1);
+
+        PoolDTO pool = new PoolDTO()
```

Comment:
Check Pools.random()

Reply:
Ended up having to add some new functions there, but this has been moved.